Src/Base/AMReX_Algorithm.H

```diff
@@ -157,6 +157,58 @@ namespace amrex
         return hi;
     }
 
+    template<typename ItType, typename ValType>
+    AMREX_GPU_DEVICE
```

Comment:
Could we make it `AMREX_GPU_HOST_DEVICE`?

Reply:
sure!